x-pack/plugins/security/public/management/roles/edit_role/validate_role.ts

```diff
@@ -72,7 +76,17 @@ export class RoleValidator {
         )
       );
     }
-    if (!role.name.match(NAME_REGEX)) {
+    if (this.buildFlavor === 'serverless' && !role.name.match(SERVERLESS_NAME_REGEX)) {
+      return invalid(
+        i18n.translate(
+          'xpack.security.management.editRole.validateRole.serverlessNameAllowedCharactersWarningMessage',
+          {
+            defaultMessage:
+              'Name must contain only letters, numbers, punctuation and printable symbols.',
+          }
+        )
+      );
+    } else if (!role.name.match(NAME_REGEX)) {
```

Comment:
Apologies in advance 😅 
It will still hit this case in serverless if it passes the serverless regex, and then still test against the non-serverless regex. I should have realized that when I suggested just the `else if`.
```suggestion
    } else if (this.buildFlavor !== 'serverless' && !role.name.match(NAME_REGEX)) {
```

This is an alternative, using only one if and i18n formatting:
```
 if (!role.name.match(this.buildFlavor === 'serverless' ? SERVERLESS_NAME_REGEX : NAME_REGEX)) {
      return invalid(
        i18n.translate(
          'xpack.security.management.editRole.validateRole.nameAllowedCharactersWarningMessage',
          {
            defaultMessage: `Name must contain only letters, numbers, {serverless, select, 0{spaces, } other{}}punctuation and printable symbols.`,
            values: { serverless: this.buildFlavor === 'serverless' },
          }
        )
      );
    }
```

Reply:
>  if (!role.name.match(this.buildFlavor === 'serverless' ? SERVERLESS_NAME_REGEX : NAME_REGEX)) 

This doesn't test the string against the regex here - so i'll use the else if case instead.